providers-sdk/v1/upstream/health/errors.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package health
+
+import (
+	"context"
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"go.mondoo.com/cnquery/v9/cli/config"
+	"go.mondoo.com/cnquery/v9/providers-sdk/v1/upstream"
+	"go.mondoo.com/ranger-rpc"
+	"runtime/debug"
+)
+
+//go:generate protoc --proto_path=. --go_out=. --go_opt=paths=source_relative --rangerrpc_out=. errors.proto
+
+func ReportPanic(product, version, build string) {
+	if r := recover(); r != nil {
+		sendPanic(product, version, build, r, debug.Stack())
+
+		// output error to console
+		panic(r)
+	}
+}
+
+// sendPanic sends a panic to the mondoo platform for further analysis if the
+// service account is configured.
+// This function does not return an error as it is not critical to send the panic to the platform.
+func sendPanic(product, version, build string, r any, stacktrace []byte) {
+	// 1. read config
+	opts, err := config.Read()
+	if err != nil {
+		log.Info().Err(err).Msg("failed to read config")
+		return
+	}
+
+	serviceAccount := opts.GetServiceCredential()
+	if serviceAccount == nil {
+		log.Info().Msg("no service account configured")
+		return
+	}
+
+	// 2. create local support bundle
+	event := &SendErrorReq{
+		ServiceAccountMrn: opts.ServiceAccountMrn,
+		AgentMrn:          opts.AgentMrn,
+		Product: &ProductInfo{
+			Name:    product,
+			Version: version,
+			Build:   build,
+		},
+		Error: &ErrorInfo{
+			Message:    "panic: " + fmt.Sprintf("%v", r),
+			Stacktrace: string(stacktrace),
+		},
+	}
+
+	// 3. send error to mondoo platform
+	proxy, err := config.GetAPIProxy()
+	if err != nil {
+		log.Info().Err(err).Msg("failed to parse proxy setting")
```

Comment:
all of those feel like they should be error level, not info

Reply:
oh yes, forgot this to change back during testing